src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionFactory.cs

```diff
@@ -417,46 +405,36 @@ static bool IsAsyncMethod(MethodInfo method)
                 // No default either. Leave it empty.
                 return null;
             };
-
-            string? description = parameter.GetCustomAttribute<DescriptionAttribute>(inherit: true)?.Description;
-            return new AIFunctionParameterMetadata(parameter.Name)
-            {
-                Description = description,
-                HasDefaultValue = parameter.HasDefaultValue,
-                DefaultValue = parameter.HasDefaultValue ? parameter.DefaultValue : null,
-                IsRequired = !parameter.IsOptional,
-                ParameterType = parameter.ParameterType,
-            };
         }
 
         /// <summary>
         /// Gets a delegate for handling the result value of a method, converting it into the <see cref="Task{FunctionResult}"/> to return from the invocation.
         /// </summary>
-        private static Type GetReturnMarshaller(MethodInfo method, out Func<object?, ValueTask<object?>> marshaller)
+        private static Func<object?, ValueTask<object?>> GetReturnMarshaller(MethodInfo method, out Type returnType)
         {
             // Handle each known return type for the method
-            Type returnType = method.ReturnType;
+            returnType = method.ReturnType;
```

Comment:
Can this be moved to the end just before the final return?

Reply:
It's being used in a bunch of locations inside the method body. Obviously I could just query `method.ReturnType` directly but that just translates to more virtual calls.